flax/serialization.py

```diff
@@ -124,7 +153,12 @@ def _dict_state_dict(xs: Dict[str, Any]) -> Dict[str, Any]:
 
 
 def _restore_dict(xs, states: Dict[str, Any]) -> Dict[str, Any]:
-  return {key: from_state_dict(value, states[str(key)])
+  diff = set(map(str, xs.keys())).difference(states.keys())
```

Comment:
we might need to be careful about being too strict with dict restores in cases of partial restorations